openmrs/src/main/java/org/motechproject/openmrs/tasks/OpenMRSTaskDataProvider.java

```diff
@@ -219,22 +219,33 @@ private ProgramEnrollmentListResult getProgramEnrollments(String lookupName, Map
             }
         }
 
-        ProgramEnrollmentListResult filteredProgramEnrollments = prepareProgramEnrollmentListResult(programEnrollments, lookupFields);
+        ProgramEnrollmentListResult filteredProgramEnrollments = prepareProgramEnrollmentListResult(programEnrollments, lookupFields, lookupName);
 
         return filteredProgramEnrollments;
     }
 
-    private ProgramEnrollmentListResult prepareProgramEnrollmentListResult(List<ProgramEnrollment> programEnrollments, Map<String, String> lookupFields) {
+    private ProgramEnrollmentListResult prepareProgramEnrollmentListResult(List<ProgramEnrollment> programEnrollments, Map<String, String> lookupFields, String lookupName) {
         ProgramEnrollmentListResult result = new ProgramEnrollmentListResult();
         String programName = lookupFields.get(PROGRAM_NAME);
-        String patientUuid = lookupFields.get(PATIENT_UUID);
 
         result.setResults(filterPrograms(programEnrollments, programName, lookupFields.get(ACTIVE_PROGRAM)));
 
         if (result.getNumberOfPrograms() > 1) {
-            LOGGER.warn(String.format("Multiple program enrollment found with the patient UUID \"%s\" and program name" +
-                    " \"%s\".", patientUuid, programName));
+            switch(lookupName) {
```

Comment:
I think that we can move this check to the method above. This method should only filter Program Enrollments.